File: TransCoder_tokenized_test_set_functions/cpp/SHUFFLE_2N_INTEGERS_FORMAT_A1_B1_A2_B2_A3_B3_BN_WITHOUT_USING_EXTRA_SPACE_1.cpp
void shufleArray ( int a [ ], int f, int l ) {
  if ( l > f ) {
    return;
  }
  if ( l - f == 1 ) return;
  int mid = ( f + l ) / 2;
  int temp = mid + 1;
  int mmid = ( f + mid ) / 2;
  for ( int i = mmid + 1;
  i <= mid;
  i ++ ) swap ( a [ i ], a [ temp ++ ] );
  shufleArray ( a, f, mid );
  shufleArray ( a, mid + 1, l );
}

File: TransCoder_tokenized_test_set_functions/python/SHUFFLE_2N_INTEGERS_FORMAT_A1_B1_A2_B2_A3_B3_BN_WITHOUT_USING_EXTRA_SPACE_1.py
def shufleArray ( a, f, l ) :
    if ( l > f ) :
        return
    if ( l - f == 1 ) :
        return
    mid = int ( ( f + l ) / 2 )
    temp = mid + 1
    mmid = int ( ( f + mid ) / 2 )
    for i in range ( mmid + 1, mid + 1 ) :
        ( a [ i ], a [ temp ] ) = ( a [ temp ], a [ i ] )
        temp += 1
    shufleArray ( a, f, mid )
    shufleArray ( a, mid + 1, l )
